scapy/fields.py

```diff
@@ -561,6 +561,8 @@ class XStrField(StrField):
     """
 
     def i2repr(self, pkt, x):
+        if not x:
```

Comment:
It seemed confusing to me that empty strings would be shown like this:
```
   |  challengelen= 16
   |  sid       =
```

Now they appear like this:
```
   |  challengelen= 16
   |  sid       = ''
```
Also, I'm not sure there's a risk for a None situation, but now it would be caught anyway.